Add controller argument to SubmenuButton

gspencergoog opened the pull request:
## Description

This adds an optional argument to the `SubmenuButton` that allows the creator to supply a `MenuController` for controlling the menu.

## Related Issues
 - Fixes https://github.com/flutter/flutter/issues/124988

## Tests
 - Added tests for new argument.

gspencergoog committed:
Add controller argument to SubmenuButton

gspencergoog committed:
Add tests

gspencergoog committed:
Amend tests, fix bug

gspencergoog committed:
Move submenu focus to onOpen

saintmac-google reviewed (commented):
the listener on the focus node is not being removed if the focusNode is provided through the widget.
line 1793 should probably say _buttonFocusNode.removeListener(_handleFocusChange)

gspencergoog committed:
Fix focus node listener removal

gspencergoog committed:
Update the copy icon in snippets and samples to use the standard one (#123651)

## Description

This updates the copy icon for copying sample code to use `content_copy` instead of `assignment`.
Before:
<img width="369" alt="223876418-2470c4c5-abfc-4511-9762-1fd0b6e05903" src="https://user-images.githubusercontent.com/8867023/228396142-450125ee-d8cf-4ede-8545-5920eb0da99d.png">

After:
<img width="408" alt="image" src="https://user-images.githubusercontent.com/8867023/228395959-8531704a-77aa-43af-9dcf-7456bb2ed090.png">

## Related Issues
 - Fixes https://github.com/flutter/flutter/issues/122246

## Tests
 - No tests needed.

QuncCccccc commented on dev/snippets/config/skeletons/sample.html:
```diff
@@ -14,7 +14,7 @@
     <button class="copy-button-overlay copy-button" title="Copy to clipboard"
       onclick="copyTextToClipboard(longSnippet{{serial}});">
-      <i class="material-icons copy-image">assignment</i>
+      <i class="material-icons copy-image">content_copy</i>
```
Seems this and the change below is not related to this PR. And looks like this change has been merged to master:)

gspencergoog replied:
Oh!  Thanks, yes, that's not related.

QuncCccccc reviewed (commented):
Looks like one unrelated commit is accidentally added to this PR. Otherwise, LGTM!

gspencergoog committed:
Revert unrelated changes

QuncCccccc reviewed (approved):
LGTM!
